sdk/schemaregistry/azure-schemaregistry/tests/perfstress_tests/_test_base.py

```diff
@@ -0,0 +1,130 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import json
+import math
+import sys
+
+from azure_devtools.perfstress_tests import PerfStressTest
+from azure.identity import DefaultAzureCredential
+from azure.identity.aio import DefaultAzureCredential as AsyncDefaultAzureCredential
+from azure.schemaregistry import SchemaRegistryClient
+from azure.schemaregistry.aio import SchemaRegistryClient as AsyncSchemaRegistryClient
+
+
+class _SchemaRegistryTest(PerfStressTest):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+
+        self.fully_qualified_namespace = self.get_from_env(
+            "SCHEMAREGISTRY_FULLY_QUALIFIED_NAMESPACE"
+        )
+        self.group_name = self.get_from_env("SCHEMAREGISTRY_GROUP")
+        self.name = "your-schema-name"
+        self.format = "Avro"
+        self.definition = self._create_schema_definition()
+
+    def _create_schema_definition(self):
+        schema_size = self.args.schema_size
+
+        fields = []
+        schema = {
+            "type": "record",
+            "name": "example.User",
+            "fields": fields,
+        }
+
+        # 100 bytes
+        schema_no_fields_size = sys.getsizeof(json.dumps(schema, separators=(",", ":")))
+        fields.append({"name": "favor_number00000", "type": ["int", "null"]})
+        # each additional field is 50 bytes
+        schema_one_field_size = sys.getsizeof(json.dumps(schema, separators=(",", ":")))
+        field_size = schema_one_field_size - schema_no_fields_size
+
+        # calculate number of fields to add to get args.schema_size rounded down to nearest 50 multiple
+        num_fields = math.floor((schema_size - schema_no_fields_size) / field_size)
+
+        for i in range(1, num_fields):
+            num_idx = f"{i:05d}"
+            fields.append(
+                {"name": f"favo_number{num_idx}", "type": ["int", "null"]},
+            )
+        definition = json.dumps(schema, separators=(",", ":"))
+        return definition
+
+    @staticmethod
+    def add_arguments(parser):
+        super(_SchemaRegistryTest, _SchemaRegistryTest).add_arguments(parser)
+        parser.add_argument(
+            "--schema-size",
+            nargs="?",
+            type=int,
+            help="Size of a single schema. Max 1000000 bytes. Defaults to 150 bytes",
+            default=150,
+        )
+
+
+class _RegisterTest(_SchemaRegistryTest):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        # CANNOT do parallel>1 , results in Conflict
+        if self.args.parallel > 1:
+            raise ValueError(
+                "Registering multiple times in parallel results in conflict. Parallel can only be 1 for `Register`."
```

Comment:
do you know what's "conflict" here? schema definitions of schemas are of the same as defined in `_create_schema_definition`?

Reply:
if it's conflicting because of the same schema definition, we could do some randomization by using uudi when generating fields/name

Reply:
ah yes it is, just fixed! one other thing is that in my current testing tier (standard) I can only add 25 schemas. Should we try to account for this and raise and error? or should we just fail and leave the warning in the readme that the max number of schemas might be reached depending on tier?

Reply:
I personally prefer the latter one which is easier.

I would like to run the same perf against basic/standard/tier and I don't want to hard code or write some sophisticated logic to deal with different tier. If there's some constrains, I should update the input to meet the requirements as long as I can get the proper error information.

Reply:
sounds good - in that case, I'll leave this as is.